### PR TITLE
feat: itf test-framework hardening; resolve system theme; restore login toggle icons; counterparty legal form radio group

### LINKED ISSUE #994: Theme switcher: "system" option isn't resolved — Lens dashboards follow OS dark mode, rest of the app doesn't
## Description
`ThemeSwitcher()` writes the raw radio value (`"system"`) as the `<html>` class instead of resolving it against `prefers-color-scheme`. Tailwind's `darkMode: "class"` strategy only reacts to a literal `.dark` class, so choosing "System" leaves the whole Alpine/Tailwind-rendered app in light mode regardless of the OS theme.

Lens React dashboards independently re-derive the theme via `matchMedia('(prefers-color-scheme: dark)')` whenever neither `.dark` nor `.light` is on `<html>` — so Lens alone follows OS dark mode while the rest of the app stays light. Any other embedded React applet that only mirrors the literal `.dark` class (`back/web/src/platform/applet-host/react-element.ts` in iota-uz/eai) would hit the same divergence.

## Where it happens
Navbar theme switcher (3-way radio: system/light/dark) — any authenticated page. Observed downstream in iota-uz/eai, `erp.eai.uz`.

## Steps to reproduce
1. Set OS to dark mode.
2. Open the theme switcher (navbar) and select "System".
3. Observe: Lens dashboards (e.g. `/analytics/profitability`) render dark; the rest of the app (nav, other pages) stays light.

## Expected vs Actual
**Expected:** "System" applies dark styling everywhere when the OS is dark, light everywhere when the OS is light, and updates live on OS-preference change.
**Actual:** `<html class="system">` is set literally — Tailwind never sees `.dark`, so Alpine/Tailwind pages stay light. Lens resolves independently, causing per-surface inconsistency.

## Root cause
`modules/core/presentation/templates/layouts/authenticated.templ:96-116`, inline script:
```js
function changeTheme(theme) {
  root.classList.remove(previousTheme);
  if (!theme) theme = initialTheme;
  window.localStorage.setItem(THEME_STORAGE_KEY, theme);
  root.classList.add(theme)   // adds literal "system", never resolved
  previousTheme = theme;
}
```
No `matchMedia` listener exists here, so `"system"` is never translated to `light`/`dark`, and OS-preference changes while "system" is active are never picked up.

Compare `back/web/src/features/lens/App.tsx:5-25` (eai, downstream), which does it correctly — resolves via `matchMedia` and listens for `change`.

## Suggested fix
- In `changeTheme()`, when `theme === "system"`, resolve via `matchMedia('(prefers-color-scheme: dark)').matches` before `classList.add`, applying `dark`/`light` (not the literal string) to `<html>`.
- Register a `matchMedia(...).addEventListener('change', ...)` that re-resolves and re-applies while "system" stays selected, so OS toggles are picked up live.
- Keep `localStorage["iota-theme"]` storing the raw choice (`system`/`light`/`dark`) — only the *applied* DOM class needs to be the resolved value.

## Acceptance criteria
- [ ] Selecting "System" applies `dark` to `<html>` when OS is dark, `light` when OS is light.
- [ ] Toggling OS theme while "System" is selected updates Granite/Alpine live, no reload — matching Lens's existing behavior.
- [ ] Lens dashboards and Granite/Alpine pages render the same theme at all times.

### LINKED ISSUE #1045: Контрагенты: заменить select правовой формы на radio group
## Контекст
В форме создания контрагента поле «Правовая форма» сейчас отрисовано выпадающим списком. Набор вариантов короткий и фиксированный: физлицо, ООО, АО, ИП.

Поле реализовано в IOTA SDK как общий компонент `CounterpartyLegalTypeSelect` в `modules/finance/presentation/templates/components/counterparty_selects.templ`; его используют формы создания и редактирования контрагента.

## Изменение
Заменить select правовой формы на radio group / набор радиокнопок, сохранив текущие значения, локализацию, валидацию и передачу значения формы. Использовать существующий компонент radio из SDK, если он соответствует этому сценарию.

## Критерии приёмки
- В форме создания контрагента правовая форма выбирается одним понятным radio-элементом, а не выпадающим списком.
- Доступны все текущие варианты: физлицо, ООО, АО и ИП.
- Выбор передаётся в серверную форму без изменения контракта данных.
- Ошибка валидации и предвыбранное значение отображаются корректно.
- Поведение формы редактирования контрагента остаётся работоспособным; при изменении общего компонента её отображение соответствует новому контролу.

### LINKED ISSUE #1082: Login page theme-toggle icons likely invisible: Tailwind v3 group-[] variant not generated under v4
## Problem

The theme toggle on the login page, `modules/core/presentation/templates/pages/login/index.templ` lines 64, 68 and 72, shows each icon with the Tailwind v3 arbitrary-group trick:

```templ
@icons.Desktop(icons.Props{Size: "20", Class: "scale-0 duration-200 peer-checked/system:group-[]/system:scale-100"})
@icons.Sun(icons.Props{Size: "20", Variant: icons.Filled, Class: "scale-0 duration-200 peer-checked/light:group-[]/light:scale-100"})
@icons.Moon(icons.Props{Size: "20", Variant: icons.Filled, Class: "scale-0 duration-200 peer-checked/dark:group-[]/dark:scale-100", ...})
```

We're on Tailwind v4 (4.1.18), and it doesn't generate CSS for `group-[]` variants. While fixing #1081 we checked the production build of `styles/tailwind/input.css` and found no `group-[]` rules at all. So the `peer-checked:…:scale-100` override never applies, and the icons stay at `scale-0`: the round button renders but the Desktop, Sun or Moon icon inside it can't be seen.

**Not yet checked in a browser.** This conclusion comes from the markup and the generated CSS. Confirm it on `/login` before fixing.

## Same root cause as #1081

#1081 fixed this exact pattern in the templ `Checkbox` (`components/base/input/input.templ`). A checked checkbox had never shown its checkmark, for the same reason. That fix replaced the trick with plain CSS classes in `styles/tailwind/iota.css`, which works the same under v4.

## Suggested fix

- Replace the three `peer-checked/X:group-[]/X:scale-100` classes with a structure Tailwind v4 generates. One option: plain CSS in `iota.css`, e.g. `.peer\/system:checked + label svg { scale: 1 }` or named classes as in #1081. Another is a v4-supported variant on the icon itself. The icon sits inside the `label`, which follows the radio `input` as a sibling. So a `peer-checked:[&_svg]:scale-100` on the label, or a named class, reaches it without `group-[]`.
- Search the rest of the codebase for other `group-[]` uses: `rg 'group-\[\]'`. Each one is broken the same way.
- Add a regression test at the level where the bug is visible: a browser/VR check that the icon on `/login` has a non-zero rendered size.

Found during iota-uz/iota-sdk#1081.

### PR DESCRIPTION
**Pull request by diyor28**
## Why

Testing-framework hardening (from a debugging session against component controllers) plus three tracked UI/UX fixes folded into the same PR.

Closes #994
Closes #1082
Closes #1045

### itf / repo — test-framework hardening

- **`Suite.MountAll()`** — registers every controller of the compiled application on the suite router, mirroring `pkg/server`'s production route table. Component controllers were never auto-mounted; forgetting `suite.Register` produced a bare 404 with no hint. Opt-in; `Register` keeps working.
- **`TestEnvironment.CommitTx(tb)` / `FreshTx(tb)`** — formalizes the "seed through a commit, keep rollback isolation afterwards" pattern for pool-based readers (`UsePool` repositories, workers). Replacement transactions are guarded and **re-apply the configured transaction settings**.
- **`repo.GuardedTx`** — overlapping calls on one transaction fail fast with `repo.ErrTxInUse` instead of racing the single connection ("conn busy", interleaved protocol traffic). Rollback-scope transactions are wrapped automatically. The guard covers every connection-driving method (Exec/Query/QueryRow/SendBatch/CopyFrom/Begin/Prepare/Commit/Rollback), spans the full lifetime of lazy handles (`pgx.Rows` releases on Close or terminal Next, `BatchResults` on Close, a row on Scan), and savepoint transactions share the parent's guard state. `repo.NewGuardedTx(tx)` exports the guard for application code.
- **i18n assertions** — `RequireMessage` / `MessageResolves` / `RequireMessages` / `RequireMessageAllLocales` pin message ids: dotted ids must resolve through nested tables, a key missing from one locale counts as unresolved, and an unsupported locale served the bundle's default language is rejected (`fr` ≠ `en`).

### #994 — theme switcher "system" resolution

Choosing "System" applied a literal `system` class to `<html>`; Tailwind's class strategy only reacts to `.dark`, so the whole Alpine/Tailwind app stayed light regardless of OS preference while embedded dashboards resolved it themselves. The shared inline script (duplicated across the authenticated layout and the login page) moves to `assets/js/theme.js`: "system" resolves via `prefers-color-scheme`, a `matchMedia` listener re-resolves live while "system" stays selected, `localStorage["iota-theme"]` keeps the raw choice and the applied DOM class is always the resolved `light`/`dark`.

### #1082 — login theme-toggle icons

The Desktop/Sun/Moon icons relied on the Tailwind v3 `peer-checked:x:group-[]/x:scale-100` trick that v4 never emits, so they stayed at `scale 0` — the toggle rendered as an empty circle. Icons now scale via plain CSS in `iota.css` (sibling combinator from the named radio peer), the same approach that fixed the templ Checkbox in #1081. Remaining `group-[]` uses were swept: none left outside `iota.css` comments.

### #1045 — counterparty legal form as radio group

`CounterpartyLegalTypeSelect` renders as a `radio.RadioGroup` + `radio.CardItem` set instead of a dropdown: four fixed options (individual, LLC, JSC, sole proprietorship) deserve one visible choice. Same options, localization, error display and form contract — radios submit under the `name` attr (`LegalType`), other attrs (`required`) apply to each input. The unused `SelectLegalType` placeholder key is dropped from all six locales. Controller tests updated to the radio markup (create: 4 unchecked radios; edit: the saved value checked).

## Testing

- `pkg/repo/guarded_tx_test.go` — gate-based fakes, `-race`: sequential pass, concurrent fail-fast for Exec/Query/QueryRow(Scan)/Commit/Rollback/Begin/Prepare, guard held while Rows open (terminal Next and Close both release), savepoint shares parent guard.
- `pkg/itf/scope_tx_test.go` — DB-backed: MountAll renders a core route (404 without it); CommitTx makes scope writes visible to pool readers and keeps rollback isolation afterwards; FreshTx discards uncommitted writes; replacement re-applies tx settings (`SHOW lock_timeout` = 2s); scope tx is guarded; `UseTx` resolves to the replacement.
- `pkg/itf/i18n_assert_test.go` — nested-path resolution, dot-path trap, partial translation, unsupported locale `fr`.
- `modules/finance/...` controller tests updated and green (radio group renders, create/edit contract unchanged).
- `e2e/tests/core/login-theme.spec.ts` — browser-level regression for #994/#1082: system resolves against emulated OS scheme, OS changes tracked live, raw choice stored, selected icon renders at non-zero size.
- Full `modules/core/...` green with the guard active.

## Compatibility

Additive for the itf/repo APIs, with one behavioural change by design: overlapping calls on a rollback-scope transaction fail fast instead of racing (always a defect in the test or code under test). UI fixes are user-visible bug fixes; the counterparty form's data contract is unchanged.

## Summary by CodeRabbit

- **New Features**
  - “System” theme selection now follows the operating system’s light/dark preference and updates automatically when that preference changes.
  - Restored visible theme-toggle icons on the login page.
  - Counterparty legal-form selection is now presented as an accessible radio group instead of a dropdown.

- **Bug Fixes**
  - Improved theme-switching consistency across login and authenticated pages.
  - Improved reliability of transaction isolation and translation validation in automated testing.

### CODE REVIEW

File: modules/core/presentation/templates/layouts/authenticated.templ [hunk 1]
```diff
@@ -94,28 +96,7 @@ templ ThemeSwitcher() {
...
-			}
-			function onThemeChange(input) {
-				changeTheme(input.value);
-			}
-			if (radioInput) {
-				radioInput.checked = true;
-				changeTheme(initialTheme);
-			}
-		</script>
+		<script src={ themeJs }></script>
```
[the remark this answers is not in the document]
Comment (diyor28) [reply modules/core/presentation/templates/layouts/authenticated.templ:99]: Not applied as suggested: in this templ version `script src` is a plain string attribute — `templ.URL()` returns `templ.SafeURL`, which does not compile there (`cannot use templ.SafeURL as string value`). Dynamic script sources elsewhere follow the same plain-string form (e.g. `src={ alpine }` in `layouts/base.templ`), so `src={ themeJs }` matches the established convention. `templ.URL` is kept for href-style attributes.
